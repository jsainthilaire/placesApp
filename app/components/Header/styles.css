.container {
    align-items: center;
    background-color: #8D6E63;
    box-shadow: 0 2px 5px rgba(0,0,0,.26);
    color: white;
    display: flex;
    font-size: 18px;
    justify-content: space-between;
    width: 100%;
}

.menuItems {
    list-style: none;
    margin-right: 12px;
}

.btn {
    border-radius: 2px;
    box-shadow: 0 1px 4px rgba(0, 0, 0, .6);
    cursor: pointer;
    line-height: 2em;
    padding: 0 20px;
}

.btn:hover, .btn:focus {
    background-color: #7d655d;
}

.btn:active {
    transform: scale(0.9, 0.9);
}

.headerLogo {
    height: 50px;
}

.link {
    color: inherit;
    text-decoration: none;
}
