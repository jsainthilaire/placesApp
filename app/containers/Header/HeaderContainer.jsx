import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Header } from 'components'
import { isAuthenticated } from 'helpers/auth'

class HeaderContainer extends Component {
  render() {
    const { status, uid } = this.props
    const isAuthed = isAuthenticated(status, uid)

    return (
      <Header isAuthed={isAuthed} />
    )
  }
}

HeaderContainer.propTypes = {
  status: PropTypes.string.isRequired,
  uid: PropTypes.string.isRequired,
}

const mapStateToProps = ({ auth }) => ({
  status: auth.status,
  uid: auth.uid,
})

export default connect(mapStateToProps)(HeaderContainer)
