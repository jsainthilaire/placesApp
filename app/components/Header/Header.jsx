import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import logo from 'assets/images/logo.png'
import { container, menuItems, btn, headerLogo, link } from './styles.css'

const AuthMenuItem = ({ isAuthed }) => {
  if (isAuthed) {
    return <li className={btn}><Link className={link} to='/'>Logout</Link></li>
  }

  return <li className={btn}><Link className={link} to='/'>Login</Link></li>
}

AuthMenuItem.propTypes = {
  isAuthed: PropTypes.bool.isRequired,
}

const Header = ({ isAuthed }) => (
  <header className={container}>
    <Link className={headerLogo} to='/'><img alt='logo' src={logo} /></Link>
    <nav>
      <ul className={menuItems}>
        <AuthMenuItem isAuthed={isAuthed} />
      </ul>
    </nav>
  </header>
)

Header.propTypes = {
  isAuthed: PropTypes.bool.isRequired,
}

export default Header
