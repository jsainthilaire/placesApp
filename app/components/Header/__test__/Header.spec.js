import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../Header'

test('Header render correctly', () => {
  const wrapper = shallow(<Header isAuthed={false} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
