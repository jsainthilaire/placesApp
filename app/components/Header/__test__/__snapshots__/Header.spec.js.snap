// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="container"
>
  <Link
    className="headerLogo"
    replace={false}
    to="/"
  >
    <img
      alt="logo"
      src="logo.png"
    />
  </Link>
  <nav>
    <ul
      className="menuItems"
    >
      <AuthMenuItem
        isAuthed={false}
      />
    </ul>
  </nav>
</header>
`;
